docs/SETUP_PROVISIONING.md

```diff
@@ -0,0 +1,1505 @@
+
+The continuous integration part of the Gitlab pipeline is not dependent on the AWS deployed infrastructure and can be run anytime.
+The provisioning of a new AWS infrastructure and deployment of Gigadb applicaiton onto it requires 
+orchestration between the Gitlab pipeline execution and the execution of the provisioning tools here, especially the Ansible playbook.
+
+There are 5 phases of provisioning, and below they are listed alongside their relationship with Gitlab
+
+| phase | description | operator actions | required Gitlab pipeline state |
+| -- | -- | -- | -- |
+| 1.TF init | The local terraform state is initiliased from a remote Gitlab state (which is created on the fly if it doesn't exist) | runs `tf_init.sh` | no state required |
+| 2.TF provisioning | The resources of the infrastructure are created on AWS | runs `terraform plan`, `terraform apply` and `terrafrom refresh` | no state required | 
+| 3.Ansible init | The AWS resources details are uploaded to Gitlab variables, Ansible dependency are built, and playbook copied to the deployment environment directory | runs `ansible_init.sh`  |no state required |
+| 4.Ansible provisioning | Configuration of the various GigaDB subsystems on each hosts of the infrastructure | executes the Ansible playbook | the build stage for the target environment should have run successfully | 
+| 5.Application deployment | deployment of the application code from git repository | none on staging, manually trigger Gitlab deployment stage on live | the deployment stage of the target environment has been triggered and ran successfully |
+
+>**Note:** You cannot run the build stage until you have run `ansible_init.sh`, because the build stage needs AWS resources details uploaded by that script
+
+>**Note:** The above implies that you cannot run the Ansible playbooks until you have run the build stage of the Gitlab pipeline, 
+
+>**Note:** GigaDB application is made of web-based and backoffice subsystems. Their deployment requires the build stage of the Gitlab pipeline to have been peformed frist.
+>Their deployment is performed using the deployment stage job (`sd_gigadb` or `ld_gigadb`) of the Gitlab pipeline for the web-based application 
+>and using the Ansible playbook  `bastion_playbook.yml` for the backoffice subsystems.
+
+## Provisioning How-to
+
+Here's a quick run-down of what you need to do to provision and infrastructure and deploy the application. 
+The remainders of sections explain in more details how to install the tools mentioned here and give additional info on the process.
+
+### Pre-requisites: Setup pipeline and Install the tools
+
+1. You have implemented the instructions in `docs/SETUP_CI_CD_PIPELINE.md`
+2. Install AWS CLI by following AWS's instructions[here](https://docs.aws.amazon.com/cli/latest/userguide/cli-chap-install.html).
+3. Install Terraform with `brew install terraform`
+4. Install Ansible with `brew install ansible`
```

Comment:
Might be worth adding a note to say ansible version 2.16 is required. @alli83 said she had to downgrade from version 2.17.